package/MDAnalysis/analysis/rms.py

```diff
@@ -507,6 +507,9 @@ def _single_frame(self):
                 self._ref_coordinates_64, self._mobile_coordinates64,
                 self._n_atoms, None, self._weights)
 
+        self._pm.format_handler = _legacy_format
```

Comment:
I think you only need to update this handler once after the PM is created. That would be after the call to super in `__init__`